Rework learning goals and make them consistent (#190)
Simulation-Software-Engineering/Lecture-Material: 02_virtualization_and_containers/README.md

Learning goals:

> - Understand the difference between virtualization and containers and when to use what.

Reviewer: I would refrain from using `understand` as a verb, as Bloom's taxonomy is anyway about different levels of _understanding_. Better:

```suggestion
- Name differences between virtualization and containers and name use cases for each.
```

- Create and modify virtual machines with VirtualBox and generate them with Vagrant.
- Create and manage Docker containers.

> - Know about Singularity.

Reviewer: ```suggestion
- Name containerization technologies beyond Docker and name their main differences.
```

Author: Do you plan to show anything beyond Docker and Singularity?

Reviewer: I would probably name at least podman as well, but the main point is that these technologies change over time.
